src/compile/axis.ts

```diff
@@ -137,6 +133,14 @@ export function parseAxis(channel: Channel, model: Model): VgAxis {
   return def;
 }
 
+function axisFieldDef(model: Model, channel: Channel): FieldDef {
```

Comment:
I made this function because we were using `model.fieldDef()` a couple times in this file which required to get the correct `channel`. I remember @kanitw mentioned we the implementation is not the best way and we would revise it.